opba-banking-protocol-facade/src/test/java/de/adorsys/opba/protocol/facade/services/authorization/ServiceSessionAndAuthSessionTest.java

```diff
@@ -0,0 +1,244 @@
+                .execute(any());
+
+        FacadeRedirectErrorResult errorResponse = (FacadeRedirectErrorResult) listAccountsService.execute(buildListAccountRequest(sessionId)).get();
+
+        assertThat(UUID.fromString(errorResponse.getAuthorizationSessionId())).isEqualTo(sessionId);
+        assertThat(UUID.fromString(errorResponse.getServiceSessionId())).isEqualTo(sessionId);
+        assertThat(errorResponse.getXRequestId()).isEqualTo(REQUEST_ID);
+        assertThat(errorResponse.getRedirectionTo().toString()).isEqualTo(REDIRECT_URL_NO_OK);
+        assertThat(errorResponse.getHeaders().get(ResponseHeaders.X_ERROR_CODE)).isEqualTo(errorResult.getCode());
+        assertThat(errorResponse.getHeaders().get(ResponseHeaders.X_ERROR_MESSAGE)).isEqualTo(errorResult.getMessage());
+
+        await().atMost(Durations.ONE_SECOND)
+                .pollDelay(Durations.ONE_HUNDRED_MILLISECONDS)
+                .until(() -> authenticationSessions.findAll().iterator().hasNext());
+
+        assertAllSessions(sessionId);
+    }
+
+    @Test
+    @SneakyThrows
+    void authSession_protocolError() {
+        UUID sessionId = UUID.randomUUID();
+
+        // STEP 1
+        ValidationErrorResult validationErrorResult = buildValidationErrorResultResult();
+
+        doAnswer(invocation -> CompletableFuture.completedFuture(validationErrorResult))
+                .when(xs2aListAccountsEntrypoint)
+                .execute(any());
+
+        FacadeStartAuthorizationResult listAccountsResponse = (FacadeStartAuthorizationResult) listAccountsService.execute(buildListAccountRequest(sessionId)).get();
+
+        assertThat(UUID.fromString(listAccountsResponse.getAuthorizationSessionId())).isEqualTo(sessionId);
+        assertThat(UUID.fromString(listAccountsResponse.getServiceSessionId())).isEqualTo(sessionId);
+        assertThat(listAccountsResponse.getRedirectionTo()).isEqualTo(validationErrorResult.getRedirectionTo());
+
+        await().atMost(Durations.ONE_SECOND)
+                .pollDelay(Durations.ONE_HUNDRED_MILLISECONDS)
+                .until(() -> authenticationSessions.findAll().iterator().hasNext());
+
+        assertAllSessions(sessionId);
+
+        // STEP 2
+        ErrorResult<AuthorizationRequiredResult> errorResult = new ErrorResult<>();
+        errorResult.setCode("400");
+        errorResult.setMessage("The addressed resource is unknown relative to the TPP");
+
+        AuthorizationRequiredResult authorizationRequiredResult = buildAuthorizationRequiredResult();
+
+        doAnswer(invocation -> CompletableFuture.completedFuture(errorResult))
+                .when(xs2aUpdateAuthorization)
+                .execute(any());
+
+        FacadeRedirectErrorResult errorResponse = (FacadeRedirectErrorResult) updateAuthorizationService.execute(buildAuthRequest(listAccountsResponse)).get();
+
+        assertThat(UUID.fromString(errorResponse.getAuthorizationSessionId())).isEqualTo(sessionId);
```

Comment:
Probably declaring static final field of UUID type for errorResponse.getAuthorizationSessionId() instead of doing
UUID.fromString(...) will reduce clutter

Reply:
use one common `sessionId` for a few tests is not a good idea because I have to poll common repositories by ID. And in some cases using common ID for sessions could have side-effects.